### ISSUE TITLE
"Unknown" date, caused by maven 504

### ISSUE THREAD

**Issue by holly-cummins** [quarkusio/extensions#75]
We query maven solr search to get extension release dates, but it seems to return 504 a high proportion of the time. That means even with retries, some extensions with otherwise-good metadata have an unknown release date. 

<img width="359" alt="image" src="https://user-images.githubusercontent.com/11509290/214364360-4c1765aa-8426-48ef-b5e1-d2ea61ba4c02.png">

It also slows down the build in an annoying way. 

I can't see an easy fix, other than trying to get the release date out of the registry, or using a maven mirror.

**Comment by holly-cummins** [comment]
Another option is using the GitHub Actions cache capability. We'd need to figure out/write an integration with the Gatsby logic
.

**Comment by holly-cummins** [comment]
Another option is to use the direct links (such as https://repo1.maven.org/maven2/io/quarkus/quarkus-core/3.0.0.Alpha3/), and use the the last-modified header sent by the server.

**Comment by holly-cummins** [comment]
This issue is further complicated by the fact that my local build gets errors ... some of the time. The CI build doesn't seem to get anything *but* errors, so we don't have any date information. I'm not seeing errors in the log, so I'm not totally sure what's happening. (We do suppress 429s because they're so common.)

**Comment by holly-cummins** [comment]
https://stackoverflow.com/questions/69790966/api-metadata-to-find-the-release-date-of-a-maven-artifact has a good discussion of options.

**Comment by gsmet** [comment]
I must say I prefer my option :]

**Comment by holly-cummins** [comment]
Interestingly, the two dates are close, but not exactly the same. For our purposes, it probably doesn't matter, since we only want the date.

```
Checking dates for  https://repo1.maven.org/maven2/io/quarkus/quarkus-smallrye-reactive-messaging-kafka/3.0.0.Alpha4/
Last modified header on the artifact directory is Thu, 09 Feb 2023 15:18:15 GMT
Modification timestamp in the json is 2023-02-09T12:43:38.000Z
```

I'm also seeing some errors coming back from http://repo1.maven.org, so I may still need to do some caching. I think I'll also do one approach and fall back to the others, in case http://search.maven.org and http://repo1.maven.org have different rate limiters and stability profiles.

**Comment by holly-cummins** [comment]
Bah. On my first browse of the deployed site, I saw a lot more timestamps. And I saw few errors in the deploy logs. But I've just noticed that https://quarkus.io/extensions/io.quarkiverse.artemis/quarkus-artemis-jms is missing a timestamp, as is https://quarkus.io/extensions/io.quarkiverse.arthas/quarkus-arthas. So we still have a problem. :( 

Re-opening. I didn't retry six times for the header like I did for the API call, so I could put in some more retries. I'm also seeing that the problems are sometimes further upstream, when I try and validate the url exists, and the `url-exist` linrary gets a 'nope, never heard of this url' reply from maven. So I'd need some retries there as well. 

Or I could just cache somewhere, to avoid putting more load on a maven central system which is clearly feeling the strain.
